### ISSUE TITLE
Still not compatible with Iris because Mixin apply for the mod failed

### ISSUE THREAD

**Issue by NaiNonTH** [kerudion/chunksfadein#18]
## Technical Details
Mod Version: 2.0.1
Minecraft Version: 1.19.3
OS: 64-bit Windows 11 Pro
Java Version: Follows Minecraft Java Execution

## Highlighted Crash Error
**Line 367-...**
```
[22:51:04] [Render thread/ERROR]: Mixin apply for mod chunksfadein failed chunksfadein.mixins.json:iris.IrisChunkShaderInterfaceMixin from mod chunksfadein -> net.coderbot.iris.compat.sodium.impl.shader_overrides.IrisChunkShaderInterface: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException Invalid descriptor on chunksfadein.mixins.json:iris.IrisChunkShaderInterfaceMixin from mod chunksfadein->@Inject::modifyConstructor(ILnet/coderbot/iris/compat/sodium/impl/shader_overrides/ShaderBindingContextExt;Lnet/coderbot/iris/pipeline/SodiumTerrainPipeline;ZLnet/coderbot/iris/gl/blending/BlendModeOverride;Ljava/util/List;FLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V! Expected (ILnet/coderbot/iris/compat/sodium/impl/shader_overrides/ShaderBindingContextExt;Lnet/coderbot/iris/pipeline/SodiumTerrainPipeline;ZLnet/coderbot/iris/gl/blending/BlendModeOverride;Ljava/util/List;FLnet/coderbot/iris/uniforms/custom/CustomUniforms;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V but found (ILnet/coderbot/iris/compat/sodium/impl/shader_overrides/ShaderBindingContextExt;Lnet/coderbot/iris/pipeline/SodiumTerrainPipeline;ZLnet/coderbot/iris/gl/blending/BlendModeOverride;Ljava/util/List;FLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V [INJECT Applicator Phase -> chunksfadein.mixins.json:iris.IrisChunkShaderInterfaceMixin from mod chunksfadein -> Apply Injections ->  -> Inject -> chunksfadein.mixins.json:iris.IrisChunkShaderInterfaceMixin from mod chunksfadein->@Inject::modifyConstructor(ILnet/coderbot/iris/compat/sodium/impl/shader_overrides/ShaderBindingContextExt;Lnet/coderbot/iris/pipeline/SodiumTerrainPipeline;ZLnet/coderbot/iris/gl/blending/BlendModeOverride;Ljava/util/List;FLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V]
org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Invalid descriptor on chunksfadein.mixins.json:iris.IrisChunkShaderInterfaceMixin from mod chunksfadein->@Inject::modifyConstructor(ILnet/coderbot/iris/compat/sodium/impl/shader_overrides/ShaderBindingContextExt;Lnet/coderbot/iris/pipeline/SodiumTerrainPipeline;ZLnet/coderbot/iris/gl/blending/BlendModeOverride;Ljava/util/List;FLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V! Expected (ILnet/coderbot/iris/compat/sodium/impl/shader_overrides/ShaderBindingContextExt;Lnet/coderbot/iris/pipeline/SodiumTerrainPipeline;ZLnet/coderbot/iris/gl/blending/BlendModeOverride;Ljava/util/List;FLnet/coderbot/iris/uniforms/custom/CustomUniforms;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V but found (ILnet/coderbot/iris/compat/sodium/impl/shader_overrides/ShaderBindingContextExt;Lnet/coderbot/iris/pipeline/SodiumTerrainPipeline;ZLnet/coderbot/iris/gl/blending/BlendModeOverride;Ljava/util/List;FLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V [INJECT Applicator Phase -> chunksfadein.mixins.json:iris.IrisChunkShaderInterfaceMixin from mod chunksfadein -> Apply Injections ->  -> Inject -> chunksfadein.mixins.json:iris.IrisChunkShaderInterfaceMixin from mod chunksfadein->@Inject::modifyConstructor(ILnet/coderbot/iris/compat/sodium/impl/shader_overrides/ShaderBindingContextExt;Lnet/coderbot/iris/pipeline/SodiumTerrainPipeline;ZLnet/coderbot/iris/gl/blending/BlendModeOverride;Ljava/util/List;FLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V]
	at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:567) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.injection.callback.CallbackInjector.inject(CallbackInjector.java:497) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.injection.code.Injector.inject(Injector.java:276) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.inject(InjectionInfo.java:445) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1378) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at io.gitlab.jfronny.libjf.unsafe.asm.AsmTransformer.transformClassBytes(AsmTransformer.java:77) ~[libjf-unsafe-v0-3.3.2-df876dbc32f0352b.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.11.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.Class.forName0(Native Method) ~[?:?]
	at java.lang.Class.forName(Class.java:467) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.MixinServiceKnot.findClass(MixinServiceKnot.java:96) ~[fabric-loader-0.14.11.jar:?]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.audit(MixinProcessor.java:254) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.audit(MixinTransformer.java:171) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.MixinEnvironment.audit(MixinEnvironment.java:1425) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at ca.fxco.memoryleakfix.MemoryLeakFix.forceLoadAllMixinsAndClearSpongePoweredCache(MemoryLeakFix.java:30) ~[memoryleakfix-1.19.3-0.7.0.jar:?]
	at net.minecraft.client.main.Main.handler$che000$loadAllMixinsThenShouldRenderAsync(Main.java:526) ~[fabric-loader-0.14.11-1.19.3.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:220) ~[fabric-loader-0.14.11-1.19.3.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[fabric-loader-0.14.11-1.19.3.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.11.jar:?]
[22:51:04] [Render thread/ERROR]: Minecraft has crashed!
net.fabricmc.loader.impl.FormattedException: java.lang.RuntimeException: Mixin transformation of net.coderbot.iris.compat.sodium.impl.shader_overrides.IrisChunkShaderInterface failed
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:463) ~[fabric-loader-0.14.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.11.jar:?]
Caused by: java.lang.RuntimeException: Mixin transformation of net.coderbot.iris.compat.sodium.impl.shader_overrides.IrisChunkShaderInterface failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427) ~[fabric-loader-0.14.11.jar:?]
.
.
.
```

## Attachment(s)
[latest.log](https://github.com/kerudion/chunksfadein/files/10315021/latest.log) [Included mods I used]

**Comment by NaiNonTH** [comment]
## Additional Info:
Iris Version: 1.5.0 beta.1

After removing Iris or Chunks Fade In mod, the game runs fine.

**Comment by kerudion** [comment]
Chunks fade in isn't compatible with iris version 1.5.0-beta.1 or later now, I'll add compatability mixins after full 1.5.0 release

**Comment by sternschnaube** [comment]
Iris 1.5 got now released.
Getting this error: https://pastebin.com/raw/gmdCu098

**Comment by kerudion** [comment]
Wow, it's a very big update! I'll work on compatibility with it now

**Comment by kerudion** [comment]
Fixed in [615a30e](https://github.com/kerudion/chunksfadein/commit/615a30eacfb223500cec133791189a76cd86982b)

**Comment by sternschnaube** [comment]
Super fast - thank you 😍👏🏿
